schematic/manifest/generator.py

```diff
@@ -1367,16 +1368,21 @@ def set_dataframe_by_url(
         Returns:
             ps.Spreadsheet: A Google Sheet object.
         """
-
         # authorize pygsheets to read from the given URL
         gc = ps.authorize(custom_credentials=self.creds)
 
         # open google sheets and extract first sheet
         # This sheet already contains headers.
         sh = gc.open_by_url(manifest_url)
         wb = sh[0]
-
-        wb.set_dataframe(manifest_df, (1, 1), fit=True)
+        try:
+            wb.set_dataframe(manifest_df, (1, 1), fit=True)
+        except HttpError as ex:
```

Comment:
Pending further discussion of whether we think this is the right approach, this new exception that's being caught should have a unit test.

This is a good example of tiny design before doing the work would be helpful, that said, sometimes you have to do a little bit before knowing your options.

Reply:
Agreed.

Reply:
Instead of an exception being caught in this way I would rather not have us modify any application code to implement a solution here.

I think the idea of using a log or span processor in the OTEL Python SDK is the appropriate solution.

The reason why I say this is: 
Using the OTEL SDK gives us one spot where all redaction logic lives, we do not need to hunt around the code base to find all the spots we need to modify. It also gives us the template to follow and apply to other projects where we want to implement similar functionality.

Additional thoughts @andrewelamb @SageGJ ?